### PR TITLE
Allow `Person` to have multiple `Survey` objects

### LINKED ISSUE #99: Extend Survey to allow a person to have multiple surveys
See discussion in #90

### PR DESCRIPTION
**Pull request by ryanlml**
Resolves #99. There were a lot more changes than I foresaw. I have changed some of the behaviour for the following commands:

- `add` now parses all `s/` prefixes.
```
add n/John Doe ... s/Survey A s/Survey B
> New person added: John Doe ... Surveys: [Survey A][Survey B]
 ```
- `view s/<KEYWORD>` no longer lists persons whose surveys contain the word `<KEYWORD>`. Instead, it lists person who has `<KEYWORD>` as a survey name. This may be improved in #107.
```
view s/Academic
> 0 persons listed!

view s/Academic Survey
> 1 persons listed!
> John Doe ... Surveys: [Academic Survey][Environment Survey]
```
- `delete s/<survey name>` now deletes persons whose surveys contain `<survey name>`, even if they contain other surveys. This is perhaps undesirable; we may want to remove the specified survey from the person rather than deleting the person altogether. Thoughts @deepimpactmir?
```
view s/Academic Survey
> 1 persons listed!
> John Doe ... Surveys: [Academic Survey][Environment Survey]

delete s/Academic Survey
> Deleted person [John Doe ... Surveys: [Academic Survey][Environment Survey]]
```
- `clone` should work as per usual

Please help test your commands! There may be other cases that messes up when multiple surveys are involved, but these are the ones I found.

Also changed the Ui to show multiple surveys, similar to `tags`.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -50,7 +50,8 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
             @JsonProperty("email") String email, @JsonProperty("address") String address,
             @JsonProperty("gender") String gender, @JsonProperty("birthdate") String birthdate,
             @JsonProperty("race") String race, @JsonProperty("religion") String religion,
-            @JsonProperty("survey") String survey, @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+            @JsonProperty("survey") List<JsonAdaptedSurvey> survey,
```
Comment (KeithPJX) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Wondering if this should be changed to surveys for both the name of the List and the JsonProperty. Will have to refactor all the JSON files though.

**Review by KeithPJX — APPROVED** [review]
LGTM. Pointed out a coding standard violation. Also, I agree that delete should just delete the survey mentioned if person has multiple surveys, if the person only has 1 survey then delete the whole person.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -59,14 +60,16 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
...
+            this.survey.addAll(survey);
+        }
         if (tagged != null) {
             this.tagged.addAll(tagged);
         }
     }
 
     /**
-     * Converts a given {@code Person} into this class for Jackson use.
+     * Converts a given {@code PJsonAdaptedTagerson} into this class for Jackson use.
```
Comment (deepimpactmir) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Not sure what this is supposed to mean. Probably a typo?

**Review by deepimpactmir — APPROVED** [review]
Thanks for all the changes! :). I agree with your sentiments on removing just the survey on persons with more than one survey. I should be able to open a PR on that soon.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 2 above)
Comment (ryanlml) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Thanks for pointing it out. Will fix them now.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 1 above)
Comment (ryanlml) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: You're right, `surveys` is better. Shouldn't be too much of a problem factoring JSON files.

Comment (ryanlml) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Renamed `survey` to `surveys` as suggested.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 2 above)
Comment (ryanlml) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Fixed typo.

**Pull request merged**